shop.py

```diff
@@ -0,0 +1,156 @@
+            else:
+                raise FundsError("3 attempts of purchases with insufficient funds.")
+
+    def remove_item(self):
+        if self.basket:
+            try:
+                response = input("Which item do you want to remove?")
+                self.basket.remove(response)
+                self.balance += self.shop.items[response]["price"]
+                self.shop.items[response]["stock"] += 1
+                print(f"{response} has been removed! Your basket is now {self.basket}, and your balance is {round(self.balance,2)}")
+            except ValueError:
+                print("Item is not in your basket. The item could not be removed.")
+        else:
+            print("No items to remove. Your basket is empty.")
+
+    def view_shelves(self):
+        for key, values in self.shop.items.items():
+            print("-", key, ": £", values['price'], "( Stock:", values['stock'],")")
+
+    def make_selection(self):
+        response = input("Make a selection:\n- View basket\n- View shelves\n- Get balance\n- Add money\n- Add item\n- Remove item\n- Checkout\n- Leave shop").lower()
+        if response == "leave shop":
+            print("Thanks for shopping!")
+            exit()
+        elif response == "view basket":
+            self.view_basket()
+        elif response == "get balance":
+            self.get_balance()
+        elif response == "add money":
+            money_wanted = input("How much money do you want to add e.g. 100?")
+            self.add_money(money_wanted)
+        elif response == "add item":
+            item_wanted = input("Which item do you want to purchase?").lower()
+            self.add_item(item_wanted)
+        elif response == "checkout":
+            self.checkout()
+        elif response == "remove item":
+            self.remove_item()
+        elif response == "view shelves":
+            self.view_shelves()
+        else:
+            print("Invalid response")
+
+    def checkout(self):
+        if self.basket:
+            print("Thank you for shopping with Tech Store! Your purchases are:")
+            for item in self.basket:
+                print(item)
+            exit()
+        else:
+            print("Your basket is empty. Either add to your basket or leave the shop.")
+
+  
+            
+
+
+
+name = input("What is your name?")
+if not name:
+    print("You must enter a name. Run again.")
+    exit()
+comp_shop = Shop("Tech Store")
+new_user = User(name, 100, comp_shop)
+comp_shop.items = {"xbox": {"price" :400, "stock": 1}, "controller": {"price": 25, "stock": 5}, "disk drive": {"price":16.99, "stock": 2}, "mouse": {"price":15.50, "stock": 3}}
+
+comp_shop.add_user(new_user)
+comp_shop.greet_user()
+inpt = input("Do you want to exit? Respond with y/n")
+comp_shop.exit_option(inpt)
+
+
+while True:
+    new_user.make_selection()
+
```

Comment:
This is such a brilliant implementation. I totally enjoyed the flow and your creativity. Keep it up Millie, you are a high flyer.